kloppy/domain/services/event_deducers/carry.py

```diff
@@ -0,0 +1,142 @@
+                            [BodyPart.HEAD, BodyPart.HEAD_OTHER]
+                        )
+                    )
+                ):
+                    continue
+
+                if hasattr(event, "end_coordinates"):
+                    last_coord = event.end_coordinates
+                elif hasattr(event, "receiver_coordinates"):
+                    last_coord = event.receiver_coordinates
+                else:
+                    last_coord = event.coordinates
+
+                new_coord = next_event.coordinates
+
+                distance_meters = pitch.distance_between(
+                    new_coord, last_coord, Unit.METERS
+                )
+                # Not far enough
+                if distance_meters < self.min_carry_length_meters:
+                    continue
+                # Too far
+                if distance_meters > self.max_carry_length_meters:
+                    continue
+
+                dt = next_event.timestamp - event.timestamp
+                # not same phase
+                if dt > self.max_carry_duration:
+                    continue
+                # not same period
+                if not event.period.id == next_event.period.id:
+                    continue
+
+                if hasattr(event, "end_timestamp"):
+                    last_timestamp = event.end_timestamp + timedelta(
+                        seconds=0.1
+                    )
+                elif hasattr(event, "receive_timestamp"):
+                    last_timestamp = event.receive_timestamp + timedelta(
+                        seconds=0.1
+                    )
+                else:
+                    last_timestamp = (
+                        event.timestamp
+                        + (next_event.timestamp - event.timestamp) / 10
+                    )
+
+                generic_event_args = {
+                    "event_id": f"{str(uuid.uuid4())}",
+                    "coordinates": last_coord,
+                    "team": next_event.team,
+                    "player": next_event.player,
+                    "ball_owning_team": next_event.ball_owning_team,
+                    "ball_state": event.ball_state,
+                    "period": next_event.period,
+                    "timestamp": last_timestamp,
+                    "raw_event": {},
+                }
+                carry_event_args = {
+                    "result": CarryResult.COMPLETE,
+                    "qualifiers": None,
+                    "end_coordinates": new_coord,
+                    "end_timestamp": next_event.timestamp,
+                }
+                new_carry = self.event_factory.build_carry(
+                    **carry_event_args, **generic_event_args
+                )
+                new_carries.append(new_carry)
+
+        for new_carry in new_carries:
+            pos = bisect.bisect_left(
+                [e.time for e in dataset.events], new_carry.time
+            )
+            dataset.records.insert(pos, new_carry)
```

Comment:
Instead of inserting the generated carries based on time, can't we insert them based on `idx` in the previous loop?